lua/toggleterm/ui.lua

```diff
@@ -178,7 +178,12 @@ function M.update_origin_window(term_window)
   if term_window ~= curr_win then origin_window = curr_win end
 end
 
-function M.scroll_to_bottom() vim.cmd("normal! G") end
+function M.scroll_to_bottom()
+  local mode = vim.api.nvim_get_mode()
+  if mode and (mode.mode == 'n' or mode.mode == 'nt') then
```

Comment:
nit: Can you rename this to something else like `info` or even `m` so it's `m.mode` or `info.mode` not `mode.mode`